src/content/guides/fine-tuning-upgrade-distruption-on-aws/index.md

```diff
@@ -0,0 +1,123 @@
+---
+title: "Fine-tuning upgrade distruption on AWS"
+description: "How configure and tune tenant clusters upgrade granuality."
+date: "2020-11-5"
+type: page
+weight: 130
+tags: ["recipe"]
+---
+
+# Fine-tuning upgrade distruption on AWS
+
+Giant Swarm release `12.7.0` allows a configuration of a parameters of [AWS CF UpdatePolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-updatepolicy.html) to have better control over the updates of a tenant cluster. This feature is currently in an `alpha`state.
+
+The configurion is done via nnotations on `AWSCluster` or `AWSMachineDeployment`. They can be added either at creation time or they can be added on already created CRs.
+
+## Max Batch Size
+
+`MaxBatchSize` specifies the maximum number of instances that AWS CloudFormation updates.
+
+This value can be set via annotation `alpha.aws.giantswarm.io/update-max-batch-size` on `AWSCluster` CR or `AWSMachineDeploymentCR`. Value set on `AWSCluster` will be used for a whole cluster. Value set on `AWSMachineDeploymentCR` will be only used for that specific NodePool and will override  value from `AWSCluster` (if set).
+
+There are two valid formats for the `MaxBatchSize`:
+
+* a integer number bigger than zero, which specifies static amount of nodes, e.g "5"
+* a decimal number in a range `0 < x <= 1.0` which defines a percentage of nodes that can be rolled in single batch, e. g. "0.25" which represents 25% of all nodes
+
+The default value is `0.3` which implies 30% of instances will be rolled in each batch.
+
+### MaxBatchSize examples
+
+Set value to static value `10` to roll maximum 10 instances per single batch.
+
+```yaml
+
+apiVersion: infrastructure.giantswarm.io/v1alpha2
+kind: AWSCluster
+metadata:
+  annotations:
+    alpha.aws.giantswarm.io/update-max-batch-size: "10"
+  labels:
+    giantswarm.io/cluster: jni9x
+    giantswarm.io/organization: giantswarm
+    release.giantswarm.io/version: 12.7.0
+  name: jni9x
+  namespace: default
+spec:
+ ....
+
+```
+
+Set value to `0.10` to roll maximum of 10% of nodes per single batch.
+
+```yaml
+
+apiVersion: infrastructure.giantswarm.io/v1alpha2
+kind: AWSCluster
+metadata:
+  annotations:
+    alpha.aws.giantswarm.io/update-max-batch-size: "0.10"
+  labels:
+    giantswarm.io/cluster: jni9x
+    giantswarm.io/organization: giantswarm
+    release.giantswarm.io/version: 12.7.0
+  name: jni9x
+  namespace: default
+spec:
+ ....
+
+```
+
+## Pause Time
+
+The amount of time that AWS CloudFormation pauses after making a change to a batch of instances to give those instances time to start software applications.
+
+This value can be set via annotation `alpha.aws.giantswarm.io/update-pause-time` on `AWSCluster` CR or `AWSMachineDeploymentCR`. Value set on `AWSCluster` will be used for a whole cluster. Value set on `AWSMachineDeploymentCR` will be only used for that specific NodePool and will override  value from `AWSCluster` (if set).
```

Comment:
```suggestion
This value can be set via annotation `alpha.aws.giantswarm.io/update-pause-time` on `AWSCluster` CR or `AWSMachineDeploymentCR`. The value set on `AWSCluster` will be used for a whole cluster. The value set on `AWSMachineDeploymentCR` will be only used for that specific NodePool and will override the value from `AWSCluster` in case it is defined.
```